tests/test_util_search.py

```diff
@@ -172,12 +172,12 @@ def test_search_return_list_globbed():
         }
     res = dpath.util.search(tdict, '/a/b/[02]')
     assert(isinstance(res['a']['b'], list))
-    assert(len(res['a']['b']) == 2)
-    assert(res['a']['b'] == [0, 2])
+    assert(len(res['a']['b']) == 3)
+    assert(res['a']['b'] == [0, None, 2])
```

Comment:
I had to re-read the docs here to make sure what this test is asserting, and I'm still not sure. At first I thought "is it intending to retrieve indexes 0 and 2", but then after reading the docs I have to assume it's matching any keys in the object that match the glob "[02]" - which appears to only happen because the keys get interpolated as strings (which is, I'm pretty sure, not our intention).

All that being said, I can't see any interpretation of util.search(x, '/a/b/[02]') where the user would expect the result to be [0, None, 2]. What's the rationale behind this change here?